build_tag.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This script checks that the current commit corresponds to a tag and then
+# builds the binaries for all the supported platforms. It is itended to
+# simplify the release process.
+
+# Get an input tag. Otherwise use the tag that corresponds to the current
+# commit. If there is no such tag then there is nothing to do.
+head=$(git rev-parse ${1:-HEAD})
+tag=$(git describe --exact-match "${head}" 2> /dev/null)
+
+bin_name="osdctl"
+
+if [ -z "${tag}" ]
+then
+  echo "Commit '${head}' doesn't correspond to any tag"
+  exit 1
+else
+  echo "Tag is '${tag}'"
+fi
+
+# This function builds for the given operating system and architecture
+# combination:
+function build_cmds {
+  # Get the parameters:
+  local os="$1"
+  local arch="$2"
+
+  # Set the environment variables that tell the Go compiler which operating
+  # system and architecture to build for:
+  export GOOS="${os}"
+  export GOARCH="${arch}"
```

Comment:
Just for sanity, can we either `unexport` these at the beginning or the end of this function?

Reply:
We might also want to exit with an error if `os` or `arch` isn't set.